src/test/compile-fail/dropck-object-cycle.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2014 The Rust Project Developers. See the COPYRIGHT
```

Comment:
I don't think this is the same as the #26656 example. That example, it seems to me, has more to do with the fact that fn pointers and trait objects allow you to "use" values of type `T` in non-trivial ways even if you have no bounds on `T`.